pkg/sql/colexec/table_function/meta_scan.go

```diff
@@ -76,7 +76,17 @@ func metaScanCall(_ int, proc *process.Process, arg *Argument) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	rbat = bats[0]
+	metaVecs := make([]*vector.Vector, len(bats[0].Vecs))
+	for i, vec := range bats[0].Vecs {
+		metaVecs[i], err = vec.Dup(proc.Mp())
+		if err != nil {
+			return false, err
+		}
+	}
+	rbat = &batch.Batch{
+		Cnt:  bats[0].Cnt,
+		Vecs: metaVecs,
+	}
 	rbat.SetAttributes(catalog.MetaColNames)
 	rbat.Cnt = 1
```

Comment:
you set `Cat = bats[0].Cnt` at line 87.  And set 1 here. Just keep only one of them.